### PR TITLE
Initial ollama implementation

### PR DESCRIPTION
**Pull request by Winston-503**
- Implement `OllamaLLM` and `OllamaLLMConfiguration`
- Manage consumptions for local models based on response metadata provided
- Update docs
- Minor refactoring

### CODE REVIEW

File: council/llm/ollama_llm.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            Consumption.token(completion_tokens, self.format_kind(TokenKind.completion)),
+            Consumption.token(prompt_tokens + completion_tokens, self.format_kind(TokenKind.total)),
+        ]
+
+    def get_ollama_duration_consumptions(self, response: Mapping[str, Any]) -> List[Consumption]:
+        if not all(key in response for key in self.DURATION_KEYS):
+            return []
+
+        # from nanoseconds to seconds
+        return [Consumption.duration(response[key] / 1e9, f"{self.model}:ollama_{key}") for key in self.DURATION_KEYS]
```
Comment (aflament) [inline council/llm/ollama_llm.py:52]: Let's get rid of `_ollama` as we are in the context of `OllamaConsumptionCalculator`

File: council/llm/ollama_llm.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            consumptions=self.to_consumptions(timer.duration, response),
+            raw_response=dict(response),
+        )
+
+    @staticmethod
+    def _build_messages_payload(messages: Sequence[LLMMessage]) -> List[Message]:
+        return [Message(role=message.role.value, content=message.content) for message in messages]
+
+    @staticmethod
+    def to_choices(response: Mapping[str, Any]) -> List[str]:
```
Comment (aflament) [inline council/llm/ollama_llm.py:118]:
```suggestion
    def _to_choices(response: Mapping[str, Any]) -> List[str]:
```

File: council/llm/ollama_llm.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        return [response["message"]["content"]]
+
+    @staticmethod
+    def to_consumptions(duration: float, response: Mapping[str, Any]) -> Sequence[Consumption]:
```
Comment (aflament) [inline council/llm/ollama_llm.py:122]:
```suggestion
    def _to_consumptions(duration: float, response: Mapping[str, Any]) -> Sequence[Consumption]:
```

File: council/llm/ollama_llm_configuration.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    def model_name(self) -> str:
+        return self._model.unwrap()
+
+    @property
+    def model(self) -> Parameter[str]:
+        """Ollama model."""
+        return self._model
+
+    @property
+    def keep_alive(self) -> Parameter[str]:
```
Comment (aflament) [inline council/llm/ollama_llm_configuration.py:55]: should'nt we prefer a float or int?

Comment (Winston-503) [reply council/llm/ollama_llm_configuration.py:55]: No, as it could be both number in seconds and duration string, such as "10m" or "24h"

**Review by aflament — APPROVED** [review]

**Pull request merged**